src/sentry/static/sentry/app/views/settings/components/forms/radioField.jsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import styled from 'react-emotion';
+import PropTypes from 'prop-types';
+
+import InputField from './inputField';
+import {growIn} from './styled/animations';
+
+class RadioField extends InputField {
+  static propTypes = {
+    id: PropTypes.number,
+    value: PropTypes.string,
+  };
+
+  onChange = (value, onChange, onBlur, e) => {
+    onChange(value, e);
+    onBlur(value, e);
+  };
+
+  isSelected = ({value, id}) => {
+    return value ? value === id : id === 0;
+  };
+
+  render() {
+    return (
+      <InputField
+        {...this.props}
+        field={({onChange, onBlur, value, disabled, ...props}) => (
+          <div>
+            {(props.choices() || []).map((choice, index) => {
+              const {id, name} = choice;
+              return (
+                <RadioLineItem
+                  key={id}
+                  onClick={this.onChange.bind(this, id, onChange, onBlur)}
+                  index={index}
+                >
+                  <RadioLineButton>
+                    {this.isSelected({value, id}) ? (
```

Comment:
You can also do this shorter version (note, make sure it's evaluating a boolean, because if it's 0, it'll render `0`):
 ```
{this.isSelected({value, id}) && <RadioLineButtonFill animate={value !== ''} />}
```